src/ddscxx/CMakeLists.txt

```diff
@@ -88,7 +88,19 @@ target_include_directories(
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/ddscxx>)
 
 set_property(TARGET ddscxx PROPERTY CXX_STANDARD 17)
-target_link_libraries(ddscxx PUBLIC CycloneDDS::ddsc)
+
+set(LINK_LIBS
+    CycloneDDS::ddsc)
+
+if (ENABLE_SHM)
+  # add iceoryx library
+  # TODO(Sumanth), this needs to be cleaned-up. Ideally the C++ language binding shouldn't
+  # depend directly on iceoryx, but should get it from Cyclone DDS
+  list(APPEND LINK_LIBS
+      iceoryx_binding_c::iceoryx_binding_c)
```

Comment:
I'd just use `target_link_libraries` on both occasions. It'll work, but it's just more complicated(?) And, we won't be able to link them in different ways and maybe confuse people in the future. Also, we probably want to link them both `PRIVATE` and the current invocation is wrong(?)

Reply:
Agreed, will also fix this